scala/scala.bzl

```diff
@@ -614,31 +670,33 @@ def _scala_binary_common(ctx, cjars, rjars, transitive_compile_time_jars, jars2l
       files=depset([ctx.outputs.executable]),
       providers = [java_provider],
       scala = scalaattr,
+      transitive_rjars = rjars, #calling rules need this for the classpath in the launcher
+      java_jar = outputs.java_jar,
       runfiles=runfiles)
 
 def _scala_binary_impl(ctx):
   jars = _collect_jars_from_common_ctx(ctx)
   (cjars, transitive_rjars) = (jars.compile_jars, jars.transitive_runtime_jars)
-  transitive_rjars += [ctx.outputs.jar]
 
+  out = _scala_binary_common(ctx, cjars, transitive_rjars, jars.transitive_compile_jars, jars.jars2labels)
   _write_launcher(
       ctx = ctx,
-      rjars = transitive_rjars,
+      rjars = out.transitive_rjars,
       main_class = ctx.attr.main_class,
       jvm_flags = ctx.attr.jvm_flags,
   )
-  return _scala_binary_common(ctx, cjars, transitive_rjars, jars.transitive_compile_jars, jars.jars2labels)
+  return out
 
 def _scala_repl_impl(ctx):
   # need scala-compiler for MainGenericRunner below
   jars = _collect_jars_from_common_ctx(ctx, extra_runtime_deps = [ctx.attr._scalacompiler])
   (cjars, transitive_rjars) = (jars.compile_jars, jars.transitive_runtime_jars)
-  transitive_rjars += [ctx.outputs.jar]
```

Comment:
this is removed because `_scala_binary_common` is putting it in also, is that right?

Reply:
yes